Xamarin.Forms.Core/Items/CarouselView.cs

```diff
@@ -1,10 +1,162 @@
-﻿using Xamarin.Forms.Platform;
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Windows.Input;
+using Xamarin.Forms.Platform;
 
 namespace Xamarin.Forms
 {
 	[RenderWith(typeof(_CarouselViewRenderer))]
 	public class CarouselView : ItemsView
 	{
+		public const string CurrentItemVisualState = "CurrentItem";
+		public const string NextItemVisualState = "NextItem";
+		public const string PreviousItemVisualState = "PreviousItem";
+		public const string VisibleItemVisualState = "VisibleItem";
+		public const string DefaultItemVisualState = "DefaultItem";
+
+		public static readonly BindableProperty PeekAreaInsetsProperty = BindableProperty.Create(nameof(PeekAreaInsets), typeof(Thickness), typeof(CarouselView), default(Thickness));
+
+		public Thickness PeekAreaInsets
+		{
+			get { return (Thickness)GetValue(PeekAreaInsetsProperty); }
+			set { SetValue(PeekAreaInsetsProperty, value); }
+		}
+
+		static readonly BindablePropertyKey VisibleViewsPropertyKey = BindableProperty.CreateReadOnly(nameof(VisibleViews), typeof(List<View>), typeof(CarouselView), null);
+
+		public static readonly BindableProperty VisibleViewsProperty = VisibleViewsPropertyKey.BindableProperty;
```

Comment:
This is not on the spec, not sure if its a good idea or not

Reply:
As we scroll the carousel, it looks at the VisibleCells,  it helps for doing the animations and setting states.